fuzzy_logic/fuzzy_operations.py

```diff
@@ -129,41 +137,128 @@ def membership(self, x: float) -> float:
+        elif self.peak < element < self.right_boundary:
+            return (self.right_boundary - element) / (self.right_boundary - self.peak)
+        elif self.peak == element:
+            return 1.0
+
+        msg = f"Invalid value {element} for fuzzy set {self}"
         raise ValueError(msg)
 
-    def union(self, other) -> FuzzySet:
+    def union(self, second_fuzzy_set) -> FuzzySet:
         """
         Calculate the union of this fuzzy set with another fuzzy set.
         Args:
-            other (FuzzySet): Another fuzzy set to union with.
+        second_fuzzy_set (FuzzySet): Another fuzzy set to union with.
         Returns:
-            FuzzySet: A new fuzzy set representing the union.
+        FuzzySet: A new fuzzy set representing the union.
 
         >>> FuzzySet("a", 0.1, 0.2, 0.3).union(FuzzySet("b", 0.4, 0.5, 0.6))
         FuzzySet(name='a ∪ b', left_boundary=0.1, peak=0.6, right_boundary=0.35)
         """
         return FuzzySet(
-            f"{self.name} ∪ {other.name}",
-            min(self.left_boundary, other.left_boundary),
-            max(self.right_boundary, other.right_boundary),
-            (self.peak + other.peak) / 2,
+            f"{self.name} ∪ {second_fuzzy_set.name}",
+            min(self.left_boundary, second_fuzzy_set.left_boundary),
+            max(self.right_boundary, second_fuzzy_set.right_boundary),
+            (self.peak + second_fuzzy_set.peak) / 2,
         )
 
-    def plot(self):
+    def plot(self) -> None:
         """
         Plot the membership function of the fuzzy set.
         """
-        x = np.linspace(0, 1, 1000)
-        y = [self.membership(xi) for xi in x]
+        x_axis = np.linspace(0, 1, 1000)
+        y_axis = [self.membership(xi) for xi in x_axis]
+
+        plt.plot(x_axis, y_axis, label=self.name)
+
+    def __contains__(self, an_element: float) -> bool:
+        """
+        Check if an element 'an_element' is a member of the fuzzy set.
+        Returns:
+        bool: True if 'x' is a member; False otherwise.
+        >>> a = FuzzySet("A", 0, 0.5, 1)
+        >>> x = 0.2
+        >>> x in a
+        True
+        >>> x = 0.6
+        >>> x in a
+        True
+        """
+        return self.membership(an_element) > 0
+
+    def __and__(self, another_fuzzy_set: FuzzySet) -> FuzzySet:
+        """
+        Calculate the intersection of this fuzzy set with another fuzzy set.
+        Args:
+        other (FuzzySet): Another fuzzy set to intersect with.
+        Returns:
+        FuzzySet: A new fuzzy set representing the intersection.
+
+        >>> a = FuzzySet("A", 0, 0.5, 1)
+        >>> b = FuzzySet("B", 0.2, 0.7, 1)
+        >>> intersection_ab = a & b
+        >>> intersection_ab.membership(0.1)
+        0.0
+        >>> intersection_ab.membership(0.35)
+        0.18749999999999994
+        """
+        intersection_name = f"{self.name} ∩ {another_fuzzy_set.name}"
+        return FuzzySet(
+            intersection_name,
```

Comment:
Just do this inline without creating a separate variable.
```suggestion
        return FuzzySet(
            f"{self.name} ∩ {another_fuzzy_set.name}",
```